tests/pragmas/tcodgendecl.nim

```diff
@@ -0,0 +1,36 @@
+discard """
+description: "Tests for the codegendecl pragma"
```

Comment:
I opted out of the `ccodeCheck` directive as that only works on the c target.

The proper approach for that is we introduce `jscodeCheck` and a `codeCheck` where the target specific ones will only run for those targets, while the general one will run for all supported targets (c and js).